cpp/src/io/json/reader_impl.cu

```diff
@@ -88,7 +95,7 @@ std::unique_ptr<table> aggregate_keys_info(std::unique_ptr<table> info)
 col_map_ptr_type create_col_names_hash_map(column_view column_name_hashes,
```

Comment:
do we still need to keep the hash map in a unique_ptr?

Reply:
https://github.com/rapidsai/cudf/blob/8f16d4c8fdbb6b0c425bbd92f253fb1f1b30c805/cpp/src/hash/concurrent_unordered_map.cuh#L135-L138

Reply:
We do need to keep it, as it is responsible for cleanup.